www/map-view-controller.js

```diff
@@ -307,6 +335,45 @@ class MapViewControllerImpl {
     );
   }
 
+  _onViewerNavigationStarted(webPayload) {
+    if (this._isNavigating) {
+      return;
+    }
+    this._isNavigating = true;
+    let externalNavigation = { messageType: "NavigationStarted", payload: webPayload};
+    Situm.updateNavigationState(externalNavigation,  () => {;}, () => {});
+  }
+
+  _onViewerNavigationUpdated(webPayload) {
+    if (!this._isNavigating) {
+      return;
+    }
+    if (webPayload.type == "PROGRESS") {
+      let externalNavigation = { messageType: "NavigationUpdated", payload: webPayload};
+      Situm.updateNavigationState(externalNavigation,  () => {}, () => {});
+    } else if (webPayload.type == "DESTINATION_REACHED") {
+      let externalNavigation = { messageType: "DestinationReached", payload: webPayload};
+      Situm.updateNavigationState(externalNavigation,  () => {}, () => {});
+      this._isNavigating = false;
+    } else {
```

Comment:
Creo que merece a pena meter un else if type == fóra de ruta. Si por algunha razón enchufan outra mensaxe, vai entrar sempre polo out of route.